### ISSUE TITLE
Global scope is ignored by block-scoped-var & no-use-before-define when VAR is used

### ISSUE THREAD

**Issue by czjvic** [eslint/eslint#3257]
```javascript
/*global a:true, b:true */
if (a === undefined) {
    var a = 12;
}

if (a === 12) {
    b = 15;
    alert('abcd');
    alert(b);
}
```

Wrong result:
```
2:5  error  a was used before it was defined     no-use-before-define
2:5  error  "a" used outside of binding context  block-scoped-var
6:5  error  "a" used outside of binding context  block-scoped-var
```
Expected result:
nothing - a is defined in global scope

Version 1.0.0

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by IanVS** [comment]
Does it make any difference if you remove the `:true` flags?

**Comment by czjvic** [comment]
```javascript
/*global a */
if (a === undefined) {
    var a = 12;
}

if (a === 12) {
    alert('abcd');
}

```
  
```
2:5  error    a was used before it was defined     no-use-before-define
2:5  error    "a" used outside of binding context  block-scoped-var
3:9  warning  "a" is read only                     no-undef
6:5  error    "a" used outside of binding context  block-scoped-var
```

it's worse...

**Comment by lo1tuma** [comment]
If `a` is a defined global, why are you trying to redeclare it using `var`?

**Comment by nzakas** [comment]
This behavior is correct. Your referencing `a` before it was defined and outside of the block in which it was defined. This code is not working the way you think it is. You probably want something more like this:

```js
/*global a */
if (window.a === undefined) {
    window.a = 12;
}

if (a === 12) {
    alert('abcd');
}
```
